addons/knowledge/static/src/components/permission_panel/permission_panel.js

```diff
@@ -181,147 +248,135 @@ class PermissionPanel extends Component {
      * @param {Proxy} member
      */
     _onRemoveMember (event, member) {
-        if (!this.state.members.includes(member)) {
-            return;
-        }
-        const willRestrict = member.based_on ? true : false;
-        const willLoseAccess = this.isLoggedUser(member);
+        const willLoseAccess = !session.is_admin && this._isLoggedUser(member);
+        const { _t } = this.env;
         const confirm = () => {
-            this.rpc({
-                route: '/knowledge/article/remove_member',
-                params: {
-                    article_id: this.props.article_id,
-                    member_id: member.based_on ? false : member.id,
-                    inherited_member_id: member.based_on ? member.id: false,
-                }
-            }).then(res => {
-                if (this._onChangedPermission(res, willLoseAccess)) {
-                    this.loadPanel();
+            const saveGuard = new Promise((resolve, reject) => {
+                if (!willLoseAccess) {
+                    resolve();
+                    return;
                 }
+                this.env.bus.trigger('save', {
+                    onSuccess: resolve,
+                    onReject: reject
+                });
+            });
+            return saveGuard.then(() => {
+                return this.rpc({
+                    route: '/knowledge/article/remove_member',
+                    params: {
+                        article_id: this.props.article_id,
+                        member_id: member.based_on ? false : member.id,
+                        inherited_member_id: member.based_on ? member.id: false,
+                    }
+                }).then(result => {
+                    if (result.error) {
+                        Dialog.alert(this, result.error, {
+                            title: _t('Error'),
+                        });
+                        return;
+                    }
+                    if (willLoseAccess) {
+                        this._redirectToHompage();
+                        return;
+                    }
+                    this._reloadPanel();
+                    if (result.reload_tree) {
+                        this.env.bus.trigger('reload_tree', {});
+                    }
```

Comment:
same kind proposal here. And this is quite the same code as in _onChangeMemberPermission
Maybe we could factorize a bit.
But I notice `this._reloadPanel();` is sometime called before sometimes after the reload_tree. even if we don't factorize, we should at least be consistant (even if it does not changes the behaviour at the end)